### PR TITLE
Base64 encode persistent keystore

### PR DESCRIPTION
**Pull request by connormullett**
**Summary of changes**
Changes introduced in this pull request:
- Use base64 to encode unencrypted private keys
- Write formatted JSON for readability when persisting the keystore.

**Reference issue to close (if applicable)**

Closes 
- #1091 

**Other information and links**

### CODE REVIEW

**Comment by connormullett** [comment]
Output of work done: 

image

File: key_management/src/keystore.rs [hunk 1]
```diff
@@ -176,15 +182,31 @@ impl KeyStore {
...
+                                .map_err(|e| {
+                                    error!(
+                                "failed to deserialize keyfile, initializing new keystore at: {:?}",
+                                file_path
+                            );
+                                    e
+                                })
+                                .unwrap_or_default();
+
+                        let mut data = HashMap::new();
```
Comment (cryptoquick) [inline key_management/src/keystore.rs:196]: let's call this `key_info`

File: key_management/src/keystore.rs [hunk 2]
```diff
@@ -325,8 +347,21 @@ impl KeyStore {
                         Ok(())
                     }
                     None => {
+                        let mut data: HashMap<String, PersistentKeyInfo> = HashMap::new();
```
Comment (cryptoquick) [inline key_management/src/keystore.rs:350]: `key_info` also here for consistency

File: key_management/src/keystore.rs [hunk 3]
```diff
@@ -489,6 +525,36 @@ mod test {
         assert!(true);
     }
 
+    #[test]
+    #[ignore = "fragile test, requires keystore.json to exist"]
+    fn test_encode_read_write() {
+        let keystore_location = PathBuf::from("/home/connor/chainsafe/forest-db");
```
Comment (cryptoquick) [inline key_management/src/keystore.rs:531]: Can you pls pls plx put these to `/tmp` ? ty

**Review by cryptoquick — APPROVED** [review]
Great solution, and I'm glad you added another test.

**Review by cryptoquick — CHANGES_REQUESTED** [review]
Also, I forgot to markt his as "Request changes", since I did leave a couple comments.

**Review by ec2 — APPROVED** [review]
Barring the requested changes from @cryptoquick , LGTM

**Review by creativcoder — APPROVED** [review]

File: key_management/src/keystore.rs (hunk 1 above)
Comment (connormullett) [reply key_management/src/keystore.rs:196]: :+1:

File: key_management/src/keystore.rs (hunk 2 above)
Comment (connormullett) [reply key_management/src/keystore.rs:350]: :+1:

**Review by cryptoquick — APPROVED** [review]
The changes look great, thanks! :+1:

**Pull request merged**